cloudfront/wellcomelibrary.org/edge-lambda/src/staticRedirects.ts

```diff
@@ -0,0 +1,26 @@
+import * as csv from '@fast-csv/parse';
+import path from "path";
+
+type RedirectRow = {
+    libraryUrl: string;
+    collectionUrl: string;
+};
+
+// TODO: Consider pre-generating JSON in build step
+export function readStaticRedirects(): Promise<Record<string, string>> {
+    const fileLocation = path.resolve(__dirname, 'redirects.csv')
+    const options = {skipLines: 1, headers: [undefined, 'libraryUrl', 'collectionUrl', undefined, undefined]}
+    return new Promise((resolve, reject) => {
+        let redirects: Record<string, string> = {}
+
+        csv.parseFile<RedirectRow, RedirectRow>(fileLocation, options)
+            .on("error", reject)
+            .on("data", (row: RedirectRow) => {
+                const lookupKey = row.libraryUrl.replace('wellcomelibrary.org', '')
+                redirects[lookupKey] = row.collectionUrl
+            })
+            .on("end", () => {
+                resolve(redirects);
+            });
+    });
+}
```

Comment:
Instead of us having to read this CSV file, we could just parse it into a ts file and include it [like this](https://github.com/wellcomecollection/wellcomecollection.org/blob/master/cache/edge_lambdas/src/redirects.ts)? Makes it way easier to validate the file in PRs etc, and ensures it's of a standard shape via types.

Just wondering if the data is faulty (returns a borked URL) - where do we deal with that? Or will we just redirect you to something that doesn't exist?

Reply:
Having this as typescript would also catch errors like duplicate keys.

Reply:
I think we should keep the CSV for ease of reading by non devs, but also generate a TS file for code validation and speed in the lambda. We should check in the build that the TS/CSV are synced.

Reply:
Could we perhaps has a TS file that generates the CSV? Eventually it will be devs predominantly adding to this, and editing CSVs is annoying and error prone.

A complete nit, and non-blocking, as we can change it later.

Reply:
I do think we should get the CSV parsing out of the lambda though.

Reply:
Interesting read on this: https://aws.amazon.com/blogs/networking-and-content-delivery/leveraging-external-data-in-lambdaedge/

Reply:
> Global variables that you declare outside of the function handler persist across multiple Lambda@Edge invocations.
😱

Reply:
>Could we perhaps has a TS file that generates the CSV? Eventually it will be devs predominantly adding to this, and editing CSVs is annoying and error prone.

The whole point is to make this readable / roughly editable by folk who don't want to read typescript, and might pass us lists of links in volume (so a csv->ts conversion is required anyway) so I think we should keep a CSV as a source.